### PR TITLE
Added tool to map textual queries to our query ID format

### PR DESCRIPTION
**Pull request by amallia**

### CODE REVIEW

File: src/map_queries.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+int main(int argc, const char **argv)
+{
+    spdlog::drop("");
+    spdlog::set_default_logger(spdlog::stderr_color_mt(""));
+
+    std::string query_filename;
+    std::string terms_filename;
+    std::optional<std::string> stopwords_filename;
+    std::optional<std::string> stemmer = std::nullopt;
+    std::string separator = "\t";
```
Comment (JMMackenzie) [inline src/map_queries.cpp:20]: What sort of input are we expecting here? I thought our queries were formatted like:
```
123:hello abc test
```

**Review by JMMackenzie — COMMENTED** [review]
Hey, looks fine, quick question about the format though :)

File: src/map_queries.cpp (hunk 1 above)
Comment (JMMackenzie) [reply src/map_queries.cpp:20]: Wait, I think I follow now.

This program reads queries of the form:
```
123:hello abc test
```

And outputs them like
```
1\t2\t3
```
where `1,2,3` are the identifiers for `hello abc test` - right?

Perhaps we should also output the query identifier?

Comment (amallia) [reply src/map_queries.cpp:20]: Yes you are right.

I can add the query identifier too

File: src/map_queries.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    using boost::adaptors::transformed;
+    using boost::algorithm::join;
+    for (auto &&q : queries) {
+        if(query_id and q.id) {
+            std::cout << *(q.id) << ":";
+        }
+        std::cout
+            << join(q.terms | transformed([](auto d) { return std::to_string(d); }), separator)
+            << std::endl;
```
Comment (elshize) [inline src/map_queries.cpp]: Flushing with `std::endl` is very inefficient, just replace with `'\n'`. Also throw a `clang-format` for a good measure.

**Review by elshize — APPROVED** [review]
Looks good. One nit pick for the future: since `\n` is one character, we can further improve performance by passing it as such `'\n'`. I don't expect it to make a difference here, though. Still, it's a good default because why not?

**Pull request merged**